core/src/main/java/com/linecorp/armeria/internal/server/websocket/DefaultWebSocketService.java

```diff
@@ -397,4 +406,9 @@ public HttpResponse encode(ServiceRequestContext ctx, WebSocket out) {
     public WebSocketProtocolHandler protocolHandler() {
         return this;
     }
+
+    @Override
+    public HttpServiceOptions options() {
```

Comment:
What do you think of setting the `HttpServiceOptions` in the `DefaultWebSocketService#failOrFallback` method like below? 

```java
private HttpResponse failOrFallback(ServiceRequestContext ctx, HttpRequest req,
                                    Supplier<HttpResponse> invalidResponse) throws Exception {
    if (fallbackService != null) {
        final HttpServiceOptions options = fallbackService.options(); 
        if (options.requestTimeoutMillis() > 0) {
            ctx.setRequestTimeoutMillis(options.requestTimeoutMillis());
        } 
        ...
        return fallbackService.serve(ctx, req);
    } else {
        return invalidResponse.get();
    }
}
```

The reason for above suggestion is because if we set `HttpServiceOption` in `WebSocketService`, handling fallback logic is separated into two parts 
- `final WebSocketUpgradeResult upgradeResult = protocolHandler().upgrade(ctx, req);` : handles upgrade and create fallbackResponse when needed 
- ```java 
  if (fallbackService() != null) {
    ctx.setRequestTimeoutMillis(fallbackService().options().requestTimeoutMillis());
      ...
  }
   ```
   => sets `HttpServiceOptions` for the fallbackResponse

Reply:
> What do you think of setting the HttpServiceOptions in the DefaultWebSocketService#failOrFallback method like below?

Oops, I missed it. Yeah we should set it in the location. 😉

Reply:
> What do you think of setting the HttpServiceOptions in the DefaultWebSocketService#failOrFallback method like below?

I agree with your idea. By the way, the code isn't checked in. Could you add the code you mentioned above?